samples/bluetooth/peripheral_uart_nrf53/network_core/src/main.c

```diff
@@ -0,0 +1,216 @@
+	printk("Connected %s\n", addr);
+
+	current_conn = bt_conn_ref(conn);
+}
+
+static void disconnected(struct bt_conn *conn, u8_t reason)
+{
+	int err;
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason %u)\n", addr, reason);
+
+	if (current_conn) {
+		bt_conn_unref(current_conn);
+		current_conn = NULL;
+	}
+
+	err = bt_nus_disconnection_evt_send(bt_conn_get_dst(conn), reason);
+	if (err) {
+		printk("disconnection event send failedi (err %d)\n", err);
+	}
+}
+
+#ifdef CONFIG_BT_GATT_NUS_SECURITY_ENABLED
+static void security_changed(struct bt_conn *conn, bt_security_t level,
+			     enum bt_security_err err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (!err) {
+		printk("Security changed: %s level %u\n", addr, level);
+	} else {
+		printk("Security failed: %s level %u err %d\n", addr, level,
+			err);
+	}
+}
+#endif
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected    = connected,
+	.disconnected = disconnected,
+#ifdef CONFIG_BT_GATT_NUS_SECURITY_ENABLED
+	.security_changed = security_changed,
+#endif
+};
+
+#if defined(CONFIG_BT_GATT_NUS_SECURITY_ENABLED)
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+}
+
+static void pairing_confirm(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	bt_conn_auth_pairing_confirm(conn);
+
+	printk("Pairing confirmed: %s\n", addr);
+}
+
+static void pairing_complete(struct bt_conn *conn, bool bonded)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing completed: %s, bonded: %d\n", addr, bonded);
+}
+
+static void pairing_failed(struct bt_conn *conn, enum bt_security_err reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing failed conn: %s, reason %d\n", addr, reason);
+}
+
+static struct bt_conn_auth_cb conn_auth_callbacks = {
+	.cancel = auth_cancel,
+	.pairing_confirm = pairing_confirm,
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed
+};
+#else
+static struct bt_conn_auth_cb conn_auth_callbacks;
+#endif
+
+static void bt_receive_cb(struct bt_conn *conn, const u8_t *const data,
+			  u16_t len)
+{
+	int err;
+	char addr[BT_ADDR_LE_STR_LEN] = {0};
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, ARRAY_SIZE(addr));
+
+	printk("Received data from: %s\n", addr);
+
+	err = bt_nus_received_evt_send(bt_conn_get_dst(conn), data, len);
+	if (err) {
+		printk("BT received event send failed (err %d)\n", err);
+	}
+}
+
+static struct bt_gatt_nus_cb nus_cb = {
+	.received_cb = bt_receive_cb,
+};
+
+void bt_ready(int err)
```

Comment:
It is intentional because there is not easy way to pass error from it to application core.
If you want I can create a new event BLE_INIT_ERROR or somethings like that and send it to application core.